apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,61 +1,45 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import {
+  Component,
+  ContentChild,
+  Directive,
+  EventEmitter,
+  Input,
+  Output,
+  TemplateRef,
+} from '@angular/core';
 import { ListItemComponent } from '../list-item/list-item.component';
 
-@Component({
-  selector: 'app-card',
-  template: `
-    <div
-      class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4"
-      [class]="customClass">
-      <img
-        *ngIf="type === CardType.TEACHER"
-        src="assets/img/teacher.png"
-        width="200px" />
-      <img
-        *ngIf="type === CardType.STUDENT"
-        src="assets/img/student.webp"
-        width="200px" />
-
-      <section>
-        <app-list-item
-          *ngFor="let item of list"
-          [name]="item.firstName"
-          [id]="item.id"
-          [type]="type"></app-list-item>
-      </section>
+interface HasId {
+  id: string;
+}
 
-      <button
-        class="rounded-sm border border-blue-500 bg-blue-300 p-2"
-        (click)="addNewItem()">
-        Add
-      </button>
-    </div>
-  `,
+@Directive({
+  selector: 'card-img',
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
 })
-export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
-  @Input() customClass = '';
-
-  CardType = CardType;
+export class CardImgDirective {}
 
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore,
-  ) {}
+@Directive({
+  selector: '[rowRef]',
+  standalone: true,
+})
+export class RowRefDirective {}
 
-  addNewItem() {
-    if (this.type === CardType.TEACHER) {
-      this.teacherStore.addOne(randTeacher());
-    } else if (this.type === CardType.STUDENT) {
-      this.studentStore.addOne(randStudent());
-    }
-  }
+@Component({
+  selector: 'app-card',
+  templateUrl: './card.component.html',
+  standalone: true,
+  host: {
+    class: 'flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4',
+  },
+  imports: [NgIf, NgFor, ListItemComponent, NgTemplateOutlet, RowRefDirective],
+})
+export class CardComponent<T extends { id: number }> {
+  @Input() list: T[] | null = null;
+  @ContentChild(RowRefDirective, { read: TemplateRef })
+  rowTemplate!: TemplateRef<{
+    $implicit: T;
+  }>;
+  @Output() addItem = new EventEmitter<void>();
```

Comment:
you can use signal queries now

Reply:
Sure! I just changed that, I'm not sure if I did it correctly tho :( If there's another way, please let me know.

Reply:
YEs that's great, you can do it for input and output as well.